bokeh/util/serialization.py

```diff
@@ -216,16 +216,16 @@ def traverse_data(obj, is_numpy=is_numpy, use_numpy=True):
         return [transform_array(el) for el in obj]
     obj_copy = []
     for item in obj:
-        if isinstance(item, (list, tuple)):
-            obj_copy.append(traverse_data(item))
-        elif isinstance(item, float):
+        if type(item) is float:
             if np.isnan(item):
                 item = 'NaN'
             elif np.isposinf(item):
                 item = 'Infinity'
             elif np.isneginf(item):
                 item = '-Infinity'
             obj_copy.append(item)
+        elif isinstance(item, (list, tuple)):
+            obj_copy.append(traverse_data(item))
```

Comment:
I 'd like to mere this for `0.12.5` but it needs a comment at the top explaining the rationale for the ordering, otherwise it risks being changed later accidentally.